tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php

```diff
@@ -341,6 +342,15 @@ private static function rebootKernel(): void
         self::$kernel->reboot($warmupDir);
     }
 
+    private static function restoreTestDB(): void
+    {
+        if (!file_exists(sprintf('%s/ps_dump_%s.sql', sys_get_temp_dir(), AppKernel::VERSION))) {
```

Comment:
It's good that this logic is now more independent, but I have one doubt though This check must be done and duplicated in numerous places, I see one problem in this if we change the place we store this file we will have to maintain (and not forget) those numerous places

I'm saying this because the current implementation has one big flow that bothers me, the dump file is built only based on the shop version. This means when you have multiple shops locally, each time you run `composer create-test-db` for different instances it overrides the same dump file (if they have the same version).

I recently modified this a bit in this PR https://github.com/PrestaShop/PrestaShop/pull/25320/files#diff-dc3d8a9437fac082be2d0d58e8069c70e9d5e14f560c6aff8c827067b0bfc9f3R61 by including the kernel version into the dump file path.

This was easily done because there were only one or two places to modify, but now it's being duplicated which will make changes in this harder to maintain.

If one day we want to include the database name inside the file path, for example, it will be more complicated.

Reply:
TLDR; I wonder if we shouldn't handle the file path generation, along with the check and exception in a dedicated class or function, instead of duplicating it everywhere it is used?

Maybe two methods:
- `DatabaseDump::getTestDumpFilePath()`
- `DatabaseDump::checkTestDumpFile()`

This class is only used by tests anyway, which could even make it a candidate to be moved into the Tests folder instead of the install one (it is never used by install component).

Reply:
Or maybe even simpler, the check could be done inside `DatabaseDump::restoreDb()` since only tests environment use it anyway, it is always bound to the `create-test-db` command anyway